app/main.py

```diff
@@ -31,14 +35,22 @@ def main() -> None:
         Message(toilet_id, MessageType.CLEAN),
     ]
 
-    # run the programs
-    service.run_program(wake_up_program)
-    service.run_program(sleep_program)
+    async def run_sequence(*functions: Awaitable[Any]) -> None:
+        for function in functions:
+            await function
+
+    async def run_parallel(*functions: Awaitable[Any]) -> None:
+        await asyncio.gather(*functions)
+
+    await run_sequence(
+        run_parallel(service.run_program(wake_up_program)),
```

Comment:
Are you sure you can run this program in parallel? Play song command may start earlier than speaker is turned on. Check the hint